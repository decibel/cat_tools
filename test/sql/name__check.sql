\set ECHO none

\i test/pgxntool/setup.sql

\set s cat_tools
\set f name__check

SELECT plan(4);

SELECT lives_ok(
  format(
    $$SELECT %I.%I( %L )$$
    , :'s', :'f'
    , 'x'
  )
  , 'Simple name'
);
SELECT lives_ok(
  format(
    $$SELECT %I.%I( %L )$$
    , :'s', :'f'
    , 'a b'
  )
  , 'Name with spaces'
);
SELECT lives_ok(
  format(
    $$SELECT %I.%I( %L )$$
    , :'s', :'f'
    , NULL
  )
  , 'NULL'
);
SELECT throws_like(
  format(
    $$SELECT %I.%I( %L )$$
    , :'s', :'f'
    , repeat( 'x', pg_column_size( 'x'::name ) + 1 )
  )
  , '"%" becomes "%" when cast to name'
  , 'Error on overflow'
);

\i test/pgxntool/finish.sql

-- vi: expandtab ts=2 sw=2
