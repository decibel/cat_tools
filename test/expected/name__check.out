\set ECHO none
1..4
ok 1 - Simple name
ok 2 - Name with spaces
ok 3 - NULL
ok 4 - Error on overflow
# TRANSACTION INTENTIONALLY LEFT OPEN!
